### PR TITLE
boringtun 0.4.0

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

raised the [release tag missing issue](https://github.com/cloudflare/boringtun/issues/234) in the upstream

### CODE REVIEW

File: Formula/boringtun.rb [hunk 1]
```diff
@@ -18,15 +23,24 @@ class Boringtun < Formula
...
+  def caveats
+    <<~EOS
+      boringtun requires root privileges so you will need to run `sudo boringtun utun`.
+      You should be certain that you trust any software you grant root privileges.
+    EOS
+  end
+
   test do
     system "#{bin}/boringtun", "--help"
     assert_match "boringtun " + version.to_s, shell_output("#{bin}/boringtun -V").chomp
```
Comment (carlocab) [inline Formula/boringtun.rb:39]:
```suggestion
    assert_match "boringtun #{version}", shell_output("#{bin}/boringtun -V").chomp
```

Comment (chenrui333) [reply Formula/boringtun.rb]: 👍

File: Formula/boringtun.rb [hunk 2]
```diff
@@ -1,11 +1,16 @@
 class Boringtun < Formula
   desc "Userspace WireGuard implementation in Rust"
   homepage "https://github.com/cloudflare/boringtun"
-  url "https://github.com/cloudflare/boringtun/archive/v0.3.0.tar.gz"
-  sha256 "1107b0170a33769db36876334261924edc71dfc1eb00f9b464c7d2ad6d5743d3"
+  url "https://static.crates.io/crates/boringtun/boringtun-0.4.0.crate"
```
Comment (carlocab) [inline Formula/boringtun.rb:4]: Can we link to the upstream issue here?

Comment (chenrui333) [reply Formula/boringtun.rb:4]: You mean add as a code comment?

Comment (carlocab) [reply Formula/boringtun.rb:4]: Yes.

Comment (chenrui333) [reply Formula/boringtun.rb:4]: actually they just tagged the release, gonna switch back to the source tarball.

Comment (chenrui333) [reply Formula/boringtun.rb]: Good timing. :)

**Review by carlocab — APPROVED** [review]

File: Formula/boringtun.rb [hunk 3]
```diff
@@ -21,12 +21,20 @@ def install
...
-    assert_match "boringtun " + version.to_s, shell_output("#{bin}/boringtun -V").chomp
-    shell_output("#{bin}/boringtun utun -v --log #{testpath}/boringtun.log", 1)
+    assert_match "boringtun #{version}", shell_output("#{bin}/boringtun -V").chomp
+
+    output = shell_output("#{bin}/boringtun utun -v --log #{testpath}/boringtun.log 2>&1", 1)
     assert_predicate testpath/"boringtun.log", :exist?
-    boringtun_log = File.read(testpath/"boringtun.log")
-    assert_match "Success, daemonized", boringtun_log.split("/n").first
+    # requires `sudo` to start
+    assert_match "BoringTun failed to start", output
```
Comment (carlocab) [inline Formula/boringtun.rb:38]: The `sudo` requirement seems new. Is this documented upstream?

File: Formula/boringtun.rb [hunk 4]
```diff
@@ -21,12 +21,20 @@ def install
     system "cargo", "install", *std_cargo_args
```
Comment (darthShadow) [inline Formula/boringtun.rb:21]:
```suggestion
    system "cargo", "install", *std_cargo_args(path: "boringtun-cli")
```

The workspace commit after this release has broken `--HEAD` installs and requires the above change in the formula.

I am not sure whether that should be done now or later and how so figured I would point it out and leave it to you folks to decide.

Comment (carlocab) [reply Formula/boringtun.rb:21]: Does adding this work for 0.4.0?

Comment (darthShadow) [reply Formula/boringtun.rb:21]: Unfortunately, no.

Comment (darthShadow) [reply Formula/boringtun.rb:21]: The binary built is also named different: `boringtun-cli` on `HEAD` vs `boringtun` on `0.4.0`

File: Formula/boringtun.rb (hunk 3 above)
Comment (chenrui333) [reply Formula/boringtun.rb:38]: I will check on this, the `sudo` issue got raised in [this one](https://github.com/cloudflare/boringtun/issues/161)

File: Formula/boringtun.rb (hunk 4 above)
Comment (Noah-Kennedy) [reply Formula/boringtun.rb:21]: We are splitting boringtun up into multiple crates, so be advised, the binary name is indeed changing in `0.5.0`.

Comment (carlocab) [reply Formula/boringtun.rb:21]: > Unfortunately, no.

We'll have to wait to make any changes then. We'll review a pull request that uses a `build.head?` conditional to fix `--HEAD` installs in the meantime.

Comment (chenrui333) [reply Formula/boringtun.rb:21]: yeah, I can file a separate PR to address the head build thing.

Comment (chenrui333) [reply Formula/boringtun.rb:21]: head build fix PR, #96461

**Pull request closed without merging**